runtime/src/iree/hal/drivers/cuda/CMakeLists.txt

```diff
@@ -59,6 +65,8 @@ iree_cc_library(
     iree::hal::utils::resource_set
     iree::hal::utils::semaphore_base
     iree::schemas::cuda_executable_def_c_fbs
+  DEFINES
+    "IREE_HAL_DRIVER_CUDA_NCCL=${IREE_HAL_DRIVER_CUDA_NCCL_VAL}"
```

Comment:
having the preprocessor option named `IREE_HAL_CUDA_NCCL_ENABLE` will match how we name boolean flags elsewhere such as in config.h (IREE_FILE_IO_ENABLE, IREE_VM_BACKTRACE_ENABLE, etc)